### PR TITLE
Defectfix

### PR DESCRIPTION
**Pull request by junmendoza**
1. Separate the function that gets the modified nodes and updates the cached ASTs

GetDeltaAstListModified() - Get only the modified AST. 
This previously did both getting the modified AST and updating the cached AST

UpdateCachedASTFromSubtrees() - Update the cached AST

2. Update the cached ASTs only if it is not Previewing the graph

@ke-yu PTAL

### CODE REVIEW

File: src/Engine/ProtoScript/Runners/LiveRunner.cs [hunk 1]
```diff
@@ -455,135 +457,163 @@ private void SetNestedLanguageBlockASTGuids(Guid guid, List<ProtoCore.AST.Node>
...
+                        st.AstNodes.Clear();
+                        st.AstNodes.AddRange(newCachedASTList);
+                        currentSubTreeList[st.GUID] = st;
                     }
                 }
+            }
+        }
+
+
+        private IEnumerable<AssociativeNode> GetDeltaAstListModified(List<Subtree> modifiedSubTrees)
```
Comment (junmendoza) [inline src/Engine/ProtoScript/Runners/LiveRunner.cs]: This function only gets the modified ASTs

File: src/Engine/ProtoScript/Runners/LiveRunner.cs [hunk 2]
```diff
@@ -455,135 +457,163 @@ private void SetNestedLanguageBlockASTGuids(Guid guid, List<ProtoCore.AST.Node>
             }
         }
 
-
-        private IEnumerable<AssociativeNode> GetDeltaAstListModified(IEnumerable<Subtree> modifiedSubTrees)
+        private void UpdateCachedASTFromSubtrees(List<Subtree> modifiedSubTrees)
```
Comment (junmendoza) [inline src/Engine/ProtoScript/Runners/LiveRunner.cs]: This function will update the cached ASTs

File: src/Engine/ProtoScript/Runners/LiveRunner.cs [hunk 3]
```diff
@@ -455,135 +457,163 @@ private void SetNestedLanguageBlockASTGuids(Guid guid, List<ProtoCore.AST.Node>
...
-                return deltaAstList;
+                for (int n = 0; n < modifiedSubTrees.Count(); ++n)
+                {
+                    Subtree subtree = modifiedSubTrees[n];
+                    if (subtree.AstNodes == null)
+                    {
+                        continue;
+                    }
+
+                    UpdateCachedASTList(subtree, subtree.ModifiedAstNodes);
```
Comment (ke-yu) [inline src/Engine/ProtoScript/Runners/LiveRunner.cs]: Do we need to check ``subtree.ModifiedAstNodes`` is null or not?

File: src/Engine/ProtoScript/Runners/LiveRunner.cs [hunk 4]
```diff
@@ -455,135 +457,163 @@ private void SetNestedLanguageBlockASTGuids(Guid guid, List<ProtoCore.AST.Node>
...
+        public List<AssociativeNode> GetDeltaASTList(GraphSyncData syncData, bool preview = false)
         {
             csData = new ChangeSetData();
             List<AssociativeNode> finalDeltaAstList = new List<AssociativeNode>();
             finalDeltaAstList.AddRange(GetDeltaAstListDeleted(syncData.DeletedSubtrees));
             finalDeltaAstList.AddRange(GetDeltaAstListAdded(syncData.AddedSubtrees));
             finalDeltaAstList.AddRange(GetDeltaAstListModified(syncData.ModifiedSubtrees));
+            if (!preview)
+            {
+                UpdateCachedASTFromSubtrees(syncData.ModifiedSubtrees);
```
Comment (ke-yu) [inline src/Engine/ProtoScript/Runners/LiveRunner.cs]: Looks like doing preview could be in separate function?

**Comment by ke-yu** [comment]
Overall LGTM.

File: src/Engine/ProtoScript/Runners/LiveRunner.cs (hunk 3 above)
Comment (junmendoza) [inline src/Engine/ProtoScript/Runners/LiveRunner.cs]: Yes, there are instances where the UI passes a null subtree. This is code that was moved from GetDeltaAstListModified

File: src/Engine/ProtoScript/Runners/LiveRunner.cs (hunk 4 above)
Comment (junmendoza) [inline src/Engine/ProtoScript/Runners/LiveRunner.cs]: Done. Its now only called in SyncronizeInternal. thanks!

**Pull request merged**